### PR TITLE
Add loading state to button on submitting

### PR DESCRIPTION
**Pull request by DonOmalVindula**
### Purpose
$subject

### Related Issues
- https://github.com/wso2-enterprise/asgardeo-product/issues/24522

### Related PRs
- 

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: features/admin.sms-providers.v1/pages/twilio-sms-provider.tsx [hunk 1]
```diff
@@ -185,7 +189,8 @@ const TwilioSMSProvider: FunctionComponent<TwilioSMSProviderPageInterface> = (
 };
 
 TwilioSMSProvider.defaultProps = {
-    "data-componentid": "twilio-sms-provider"
+    "data-componentid": "twilio-sms-provider",
+    isLoading: false
```
Comment (pavinduLakshan) [inline features/admin.sms-providers.v1/pages/twilio-sms-provider.tsx:193]: defaultProps is going to be removed in a future react version, so let's not add new defaultProps.

File: features/admin.sms-providers.v1/pages/twilio-sms-provider.tsx [hunk 2]
```diff
@@ -41,6 +42,7 @@ const TwilioSMSProvider: FunctionComponent<TwilioSMSProviderPageInterface> = (
     const {
         ["data-componentid"]: componentId,
         onSubmit,
+        isLoading,
```
Comment (pavinduLakshan) [inline features/admin.sms-providers.v1/pages/twilio-sms-provider.tsx:45]: It'd be better to have this loading value taken from a context. Let's do it as an improvement later.

File: features/admin.sms-providers.v1/pages/custom-sms-provider.tsx [hunk 3]
```diff
@@ -228,6 +230,8 @@ const CustomSMSProvider: FunctionComponent<CustomSMSProviderPageInterface> = (
                                         onClick={ onSubmit }
                                         ariaLabel="SMS provider form update button"
                                         data-componentid={ `${componentId}-update-button` }
+                                        loading={ isLoading }
+                                        disabled={ isLoading }
```
[the remark this answers is not in the document]
Comment (DonOmalVindula) [reply features/admin.sms-providers.v1/pages/custom-sms-provider.tsx]: Fixed with 69799c37444d286ed33ae7f5f7db0f4e8be71823

File: features/admin.sms-providers.v1/pages/twilio-sms-provider.tsx (hunk 1 above)
Comment (DonOmalVindula) [reply features/admin.sms-providers.v1/pages/twilio-sms-provider.tsx]: Fixed with 69799c37444d286ed33ae7f5f7db0f4e8be71823

File: features/admin.sms-providers.v1/pages/twilio-sms-provider.tsx (hunk 2 above)
Comment (DonOmalVindula) [reply features/admin.sms-providers.v1/pages/twilio-sms-provider.tsx:45]: ack

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**